### PR TITLE
[Bug] Fix the Latest checkpoint not updated when job restart in K8S mode bug

### PR DESCRIPTION
**Pull request by youyangkou**
Problem Summary:

### What is changed and how it works?

Proposal: [the bug issue](https://github.com/streamxhub/streampark/issues/1425) 

Feel free to ping committers for the review!

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/streamxhub/streamx/issues).

  - Name the pull request in the form "[Feature] [component] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Minor fixes should be named following this pattern: `[hotfix] [docs] Fix typo in README.md doc`.

### CODE REVIEW

**Comment by youyangkou** [comment]
@wolfboys @MonsterChenzhuo PTAL

**Comment by youyangkou** [comment]
Wait a minute, there are one change that have not been submitted, I will submit them tonight

**Comment by MonsterChenzhuo** [comment]
LGTM

File: streamx-console/streamx-console-service/src/main/java/com/streamxhub/streamx/console/core/task/CheckpointProcessor.java [hunk 1]
```diff
@@ -31,10 +31,12 @@
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
+import scala.Tuple2;
+
 @Component
 public class CheckpointProcessor {
 
-    private final Map<Long, Long> checkPointCache = new ConcurrentHashMap<>(0);
+    private final Map<Tuple2<Long, String>, Long> checkPointCache = new ConcurrentHashMap<>(0);
```
Comment (wolfboys) [inline streamx-console/streamx-console-service/src/main/java/com/streamxhub/streamx/console/core/task/CheckpointProcessor.java]: Tuple2 is not recommended

File: streamx-console/streamx-console-service/src/main/java/com/streamxhub/streamx/console/core/task/CheckpointProcessor.java [hunk 2]
```diff
@@ -51,27 +53,29 @@ public void process(Long appId, CheckPoints checkPoints) {
         CheckPoints.Latest latest = checkPoints.getLatest();
         if (latest != null) {
             Application application = applicationService.getById(appId);
+            String jobId = application.getJobId();
             CheckPoints.CheckPoint checkPoint = latest.getCompleted();
```
Comment (wolfboys) [inline streamx-console/streamx-console-service/src/main/java/com/streamxhub/streamx/console/core/task/CheckpointProcessor.java:56]: you can:

String cacheId = appId + "_" + jobId;

checkPointCache.get(cacheId);
...
checkPointCache.put(cacheId, checkPoint.getId());

Comment (youyangkou) [reply streamx-console/streamx-console-service/src/main/java/com/streamxhub/streamx/console/core/task/CheckpointProcessor.java:56]: ok, i change this code

File: streamx-console/streamx-console-service/src/assembly/script/schema/pgsql-schema.sql [hunk 3]
```diff
@@ -383,6 +383,7 @@ drop table if exists "public"."t_flink_savepoint";
 create table "public"."t_flink_savepoint" (
   "id" int8 not null,
   "app_id" int8 not null,
+  "chk_id" int8 ,
```
Comment (wolfboys) [inline streamx-console/streamx-console-service/src/assembly/script/schema/pgsql-schema.sql]: int8,

**Comment by wolfboys** [comment]
LGTM

**Pull request merged**